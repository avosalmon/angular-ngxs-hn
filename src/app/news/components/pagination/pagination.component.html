<div class="container">
  <a routerLink="/news/{{ prevPage() }}" *ngIf="prevPage() > 0">< prev</a>
  <span class="disabled" *ngIf="prevPage() < 1">< prev</span>
  <span>{{ page }}</span>
  <a routerLink="/news/{{ nextPage() }}">next ></a>
</div>
