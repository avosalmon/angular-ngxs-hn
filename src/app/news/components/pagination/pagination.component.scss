.container {
  display: flex;
  justify-content: space-between;
  width: 150px;
  margin: 0 auto;
  padding: 1rem 0;
}
