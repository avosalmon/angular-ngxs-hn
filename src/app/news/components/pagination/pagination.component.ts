import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'app-pagination',
  templateUrl: './pagination.component.html',
  styleUrls: ['./pagination.component.scss']
})
export class PaginationComponent {

  @Input() page: number;

  @Output() pageChanded: EventEmitter<number> = new EventEmitter();

  onPrev() {
    if (this.page > 1) {
      const prev = this.page - 1;
      this.pageChanded.emit(prev);
    }
  }

  onNext() {
    const next = this.page + 1;
    this.pageChanded.emit(next);
  }
}
