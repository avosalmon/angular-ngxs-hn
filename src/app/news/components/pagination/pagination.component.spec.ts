import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { PaginationComponent } from './pagination.component';

describe('PaginationComponent', () => {
  let component: PaginationComponent;
  let fixture: ComponentFixture<PaginationComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [PaginationComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PaginationComponent);
    component = fixture.componentInstance;
    component.page = 1;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
