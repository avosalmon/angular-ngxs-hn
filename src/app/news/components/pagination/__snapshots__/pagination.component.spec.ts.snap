// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationComponent should match snapshot 1`] = `
<app-pagination
  page={[Function Number]}
>
  <div
    class="container"
  >
    
    
    <span
      class="disabled"
    >
      &lt; prev
    </span>
    <span>
      1
    </span>
    <a
      href="/news/2"
      ng-reflect-router-link="/news/2"
    >
      next &gt;
    </a>
  </div>
</app-pagination>
`;
